File: Database/entity/patient/Patient.go
package patient

import (
	"gorm.io/gorm"
	"time"
)

type Patient struct {
	gorm.Model
	CitizenID        string    `json:"citizenID"`
	Prefix           string    `json:"prefix"`
	FirstName        string    `json:"firstname"`
	LastName         string    `json:"lastname"`
	NickName         string    `json:"nickname"`
	CongenitaDisease string    `json:"congenitadisease"`
	BloodType        string    `json:"blood_type"`
	Gender           string    `json:"gender"`
	Birthday         time.Time `json:"birthday" gorm:"type:date"`
	PhoneNumber      string    `json:"phone_number"`
	Age              int       `json:"age"`
	DrugAllergyType  string    `json:"drug_allergy_type"`
	DrugAllergy      string    `json:"drug_allergy"`

	ContactPerson    *ContactPerson     `json:"contactperson,omitempty" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
	Address          *Address           `json:"address,omitempty" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
	Histories        []HistoryPatient   `json:"histories" gorm:"foreignKey:PatientID "`
	InitialSymptomps []InitialSymptomps `json:"initialsymptomps" gorm:"foreignKey:PatientID;references:ID"`

}

// type PatientData struct {
// 	gorm.Model
// 	CitizenID        string    `json:"citizenID"`
// 	Prefix           string    `json:"prefix"`
// 	FirstName        string    `json:"firstname"`
// 	LastName         string    `json:"lastname"`
// 	NickName         string    `json:"nickname"`
// 	CongenitaDisease string    `json:"congenitadisease"`
// 	BloodType        string    `json:"blood_type"`
// 	Gender           string    `json:"gender"`
// 	Birthday time.Time `json:"birthday" time_format:"2006-01-02" time_utc:"1" binding:"required"`
// 	PhoneNumber      string    `json:"phone_number"`
// 	Age              int       `json:"age"`
// 	DrugAllergyType string    `json:"drug_allergy_type"`
// 	DrugAllergy      string    `json:"drug_allergy"`

// 	ContactPerson *ContactPerson `json:"contactperson,omitempty" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
// 	Address       *Address       `json:"address,omitempty" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
// 	Histories   []HistoryPatient `json:"histories" gorm:"foreignKey:PatientID"`
// 	InitialSymptomps []InitialSymptomps `json:"initialsymptomps" gorm:"foreignKey:PatientID"`
// };

File: frontend/src/interface/initailPatient/patient.ts
export interface ContactPerson {
  relationship?: string;
  emergency_phone?: string;
}

export interface Address {
  house_number?: string;
  moo?: string;
  subdistrict?: string;
  district?: string;
  province?: string;
  postcode?: string;
}

export interface Patient {
  citizenID?: string;
  prefix?: string;
  firstname?: string;
  lastname?: string;
  nickname?: string;
  congenita_disease?: string;
  blood_type?: string;
  gender?: "male" | "female" | "";

  birthday?: string;      // "YYYY-MM-DD"
  phone_number?: string;
  age?: number;

  drug_allergy_type?: "hasAllergy" | "noAllergy"; // optional ถ้าอยากส่ง
  drug_allergy?: string;  // ✅ backend ต้องการ snake_case

  contactperson?: ContactPerson | null;
  address?: Address | null;
}

export interface PatientRow {
  id: string | number;
  citizenID: string;
  firstname: string;
  lastname: string;
  phonenumber: string;
}
